src/App.tsx

```diff
@@ -1,14 +1,73 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+export enum Filter {
+  All = 'all',
+  Active = 'active',
+  Completed = 'completed',
+}
+
+function filteringTodos(
+  array: Todo[],
+  settings: { filterType: Filter; query: string },
+) {
+  let arrayCopy = [...array];
+
+  if (settings.filterType === Filter.All) {
+    arrayCopy = [...array];
+  }
+
+  if (settings.filterType === Filter.Active) {
+    arrayCopy = arrayCopy.filter(element => element.completed === false);
+  }
+
+  if (settings.filterType === Filter.Completed) {
+    arrayCopy = arrayCopy.filter(element => element.completed === true);
+  }
+
+  if (settings.query) {
+    arrayCopy = arrayCopy.filter(element =>
+      element.title
+        .toLowerCase()
+        .includes(settings.query.toLowerCase().trimStart()),
+    );
+  }
+
+  return arrayCopy;
+}
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isTodosLoaded, setIsTodosLoaded] = useState(false);
+
+  const [filterType, setFilterType] = useState<Filter>(Filter.All);
+  const [query, setQuery] = useState('');
+
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [selectedUserId, setSelectedUserId] = useState<number>(0);
+
+  const handleResetQuery = () => {
+    setQuery('');
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => alert('Something went wrong, sir!!!'))
+      .finally(() => setIsTodosLoaded(true));
+  }, []);
```

Comment:
The error handling in the `useEffect` hook is not sufficient. It only shows an alert, which is not user-friendly and does not provide any information about the error. Also, it would be better to handle the loading state in a more sophisticated way. For example, you could show a loading spinner while the todos are being fetched. This will improve the user experience.